tryreduce.cc

```diff
@@ -0,0 +1,47 @@
+#ifdef _USE_OMP_
+#  include<omp.h>
+#endif
+#include<iostream>
+
```

Comment:
what is this function supposed to do?

Reply:
valgrind found some memory leaks, and I wanted to see if they were coming from my code or from somewhere else. They also appear in this minimum example.